library/cartridge_get_disabled_instances.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python
+
+from ansible.module_utils.helpers import Helpers as helpers
+
+argument_spec = {
+    'module_hostvars': {'required': True, 'type': 'dict'},
+    'play_hosts': {'required': True, 'type': 'list'},
+}
+
+
+def get_disabled_instances(instance_vars):
+    return instance_vars['instance_info']['cluster_disabled_instances']
+
+
+def get_topology_checksum(instance_vars):
+    return instance_vars['instance_info']['cluster_topology_checksum']
+
+
+def check_config_mismatch(module_hostvars, instance_name, other_hosts):
+    current_checksum = get_topology_checksum(module_hostvars[instance_name])
+
+    for other_name in other_hosts:
+        other_checksum = get_topology_checksum(module_hostvars[other_name])
+        if current_checksum != other_checksum:
+            return True
+
+    return False
+
+
+def count_disabled_instances(params):
+    module_hostvars = params['module_hostvars']
+    play_hosts = params['play_hosts']
+
+    config_mismatch_count = 0
+    started_count = 0
+    rating = {}
+
+    play_hosts = list(filter(
+        # Disabled instances is None on stateboard and not started instances
+        lambda name: all([
+            get_disabled_instances(module_hostvars[name]) is not None,
+            get_topology_checksum(module_hostvars[name]) is not None,
+        ]),
+        play_hosts,
+    ))
+
+    for instance_name in play_hosts:
+        disabled_instances = get_disabled_instances(module_hostvars[instance_name])
+
+        not_disabled_names = list(filter(lambda other_name: other_name not in disabled_instances, play_hosts))
+        if check_config_mismatch(module_hostvars, instance_name, not_disabled_names):
+            config_mismatch_count += 1
+            continue
+
+        started_count += 1
```

Comment:
Generally, an instance can be both`started` and `config_mismatch`. 
In this case, this check can be

```python
# started_count == 0 and config_mismatch_count > 0
config_mismatch_count >= started_count
```

Reply:
started_count == healthy_count. So, it's possible that 1 healthy and 3 with config mismatch. So, we should use information only from 1 healthy instance (if 3 others disabled)